components/automate-ui/src/app/modules/user/user-management/user-form/user-form.component.ts

```diff
@@ -1,11 +1,81 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit, OnDestroy, EventEmitter, Output } from '@angular/core';
 import { FormGroup } from '@angular/forms';
+import { UsernameMapper } from 'app/helpers/auth/username-mapper';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-user-form',
   templateUrl: './user-form.component.html',
   styleUrls: ['./user-form.component.scss']
 })
-export class UserFormComponent {
+export class UserFormComponent implements OnInit, OnDestroy {
   @Input() createUserForm: FormGroup;
+  @Input() conflictErrorEvent: EventEmitter<boolean>;
+  @Input() passwordErrorEvent: EventEmitter<boolean>;
+  @Input() closeEvent: EventEmitter<any>;
+
+  @Output() close = new EventEmitter();
+
+  public conflictError = false;
+  public passwordError = false;
+  public modifyUsername = false; // Whether the edit Username form is open or not.
+
+  private isDestroyed = new Subject<boolean>();
+
+  ngOnInit(): void {
+    this.conflictErrorEvent.pipe(takeUntil(this.isDestroyed))
+      .subscribe((isConflict: boolean) => {
+        this.conflictError = isConflict;
+        // Open the ID input on conflict so user can resolve it.
+        this.modifyUsername = isConflict;
+      });
+
+    this.passwordErrorEvent.pipe(takeUntil(this.isDestroyed))
+      .subscribe((badPassword: boolean) => {
+        this.passwordError = badPassword;
+        this.createUserForm.get('password').reset();
+        this.createUserForm.get('confirmPassword').reset();
+      });
+  }
+
+  ngOnDestroy() {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  handleUsernameInput(event: KeyboardEvent): void {
+    if (this.isNavigationKey(event)) {
+      return;
+    }
+    this.conflictError = false;
+  }
+
+  handleNameInput(event: KeyboardEvent): void {
```

Comment:
Please add tests for this--prior art is here: https://github.com/chef/automate/blob/8f73a3cc6296c91471b039ff3b85a1409c191d47/components/automate-ui/src/app/components/create-object-modal/create-object-modal.component.spec.ts#L49-L64